src/MPAGSCipher/PlayfairCipher.cpp

```diff
@@ -0,0 +1,175 @@
+        if (lettersFound.find(key_letter) != std::string::npos) {
+            return true ;
+        } else {
+            lettersFound += key_letter;
+            return false;
+        }
+        });
+    key_.erase(key_enc, key_.end());
+
+    // Store the coords of each letter
+    //using coord = std::pair<int, int>; (this is in .hpp file)
+    std::map <coord, char> coordToLetter ;
+    std::map <char, coord> letterToCoord ;
+    for (std::size_t i{0}; i < key_.size(); ++i) {
+        std::pair <coord, char> coordLetter { {i%5 , i/5 } , key_[i]};
+        std::pair <char, coord> letterCoord { key_[i] , {i%5 , i/5 } };
+        coordToLetter.insert( coordLetter) ;
+        letterToCoord.insert( letterCoord) ;
+    }
+
+    // Store the playfair cihper key map
+    letterToCoord_ = letterToCoord ;
+    coordToLetter_ = coordToLetter ;
+
+    // for ( auto p : letterToCoord ){
+    //     std::cout << p.first << ": [" << p.second.first << p.second.second << "]" << std::endl;
+    // }
+
+}
+
+
+std::string PlayfairCipher::applyCipher(const std::string& inputText, const CipherMode cipherMode) const
+{
+    std::string text {inputText};
+
+    // Change J → I
+    std::transform(text.begin() , text.end(), text.begin(), [] (char inLetter) {
+        if (inLetter== 'J') { inLetter = 'I' ;}
+        return inLetter ;
+    } );
+    //std::cout << "Modified input : " << text << std::endl;
+
+//    If repeated chars in a digraph add an X or Q if XX
+
+    // for (auto iter {text.begin()}; iter != text.end() ; ++iter){
+    //     if (*iter == *(iter-1)){
+    //         char charToAdd {'X'};
+    //         if (*iter == 'X'){
+    //             charToAdd='Q';
+    //         }
+    //         text.insert( iter, 1, charToAdd);
+    //         iter++ ;
+    //     }        
+    // }
+    if (cipherMode ==CipherMode::Encrypt){
+            
+        for (std::size_t i {0}; i < text.size() ; ++i){
+            if (text[i] == text[i-1]){
+                char charToAdd {'X'};
+                if (text[i] == 'X'){
+                    charToAdd='Q';
+                }
+                text.insert( i, 1, charToAdd);
+                i++ ;
+            }        
+        }
+    }
+
+    //std::cout << "Modified input : " << text << std::endl;
+
+    // if the size of input is odd, add a trailing Z
+    if (text.size()%2 == 1){
+        text += 'Z';
+    }
+    //std::cout << "Modified input : " << text << std::endl;
+
+    // Loop over the input in Digraphs
+    coord coord1; 
+    coord coord2;
+    coord newCoord1;
+    coord newCoord2;
+    std::string outputText {""};
+    for (auto txtIter {text.begin()}; txtIter != text.end() ; txtIter+=2){
+        // - Find the coords in the grid for each digraph
+        auto iter1 { letterToCoord_.find (*txtIter) };
+        auto iter2 { letterToCoord_.find (*(txtIter+1)) };
+        coord1 = (*iter1).second ; 
+        coord2 = (*iter2).second ; 
```

Comment:
Using the `.at(...)` member function of the map instead of `.find(...)` would make this a bit neater:
```suggestion
        coord1 = letterToCoord_.at (*txtIter) };
        coord2 = letterToCoord_.at (*(txtIter+1)) };
```